index.html

```diff
@@ -345,8 +345,35 @@ <h2>General rules</h2>
     </section>
 		<section id="roleMappingComputedRole">
 		  <h2>Computed Role</h2>
-                  <p>The computed role of an element is a string that represents the role of the element for the purposes of specification comformance testing. When an element has a role but is not contained in the required context (for example, a role `listitem` used on an element outside of a `list`), the computed role of the element is unspecified.</p>
-                  <p class="note">User agents can provide this role string, for example, in response to the WebDriver function <a href="https://w3c.github.io/webdriver/#get-computed-role">`getComputedRole`</a>.</p>
+      <p>The `computedrole` of an element is a string that represents the role of the element as computed by the browser engine. The `computedrole` is used primarily for the purposes of developer tools and specification comformance and interoperability testing.</p>
+      <p class="note">User agents provide this role string, for example, in developer tools, and in response to the WebDriver function <a href="https://w3c.github.io/webdriver/#get-computed-role">`getComputedRole`</a>, which is used for <a href="https://github.com/w3c/aria/blob/main/documentation/tests.md">interoperability testing of ARIA, HTML-AAM, and other specifications</a>.</p>
+      <aside class="example">
+
+<!-- ReSpec needs these examples to be unindented. -->
+<pre>&lt;button&gt; &lt;!-- computedrole returns "button" --&gt;
+
+&lt;a href="#" role="button"&gt; &lt;!-- computedrole returns "button" --&gt;</pre>
+
+      </aside>
+      <p>When an element has a role but is not contained in the required context (for example, an orphaned `listitem` without the required accessible parent of role `list`), User Agents MUST ignore the role token, and return the `computedrole` as if the ignored role token had not been included.</p>
+      <aside class="example">
+
+<!-- ReSpec needs these examples to be unindented. -->
+<pre>&lt;div role="listitem"&gt; &lt;!-- Author error: orphaned listitem. computedrole returns "generic" --&gt;
+
+&lt;div role="list"&gt; &lt;!-- computedrole returns "list" --&gt;
+  &lt;div role="listitem"&gt; &lt;!-- computedrole returns "listitem" in the required context. --&gt;</pre>
+
+      </aside>
+      <p>When host language elements do not have an exact or equivalent mapping to a concrete role, the related Accessibilty API Mapping extension specification MAY specify a unique `computedrole` string as the return value for interoperability testing purposes, such as `&lt;video&gt; -&gt; "html-video"` in [[HTML-AAM]]. However, authors MUST NOT use any non-concrete `computedrole` string in the `role` attribute, and User Agents MUST ignore any non-concrete role token.</p>
```

Comment:
only suggestion is to change "concrete role" to "existing WAI-ARIA" (or could just be "ARIA") role.  just say exactly what that is, rather than use wording that could be confused as "well what's a 'concrete' role? where is that defined?"


```suggestion
      <p>When host language elements do not have an exact or equivalent mapping to an existing WAI-ARIA role, the related Accessibilty API Mapping extension specification MAY specify a unique `computedrole` string as the return value for interoperability testing purposes, such as `&lt;video&gt; -&gt; "html-video"` in [[HTML-AAM]]. However, authors MUST NOT use any non-concrete `computedrole` string in the `role` attribute, and User Agents MUST ignore any non-concrete role token.</p>
```

Reply:
But the concrete roles aren't all defined in WAI-ARIA. The concrete `graphics-*` roles are defined in Graphics ARIA. The concrete `dpub-*` roles are defined in DPUB ARIA. Etc.

I got the term from this section of the spec... Should we define it there? https://w3c.github.io/aria/#host_general_role

Reply:
```suggestion
      <p>When host language elements do not have an exact or equivalent mapping to a valid, non-abstract role, the related Accessibilty API Mapping extension specification MAY specify a unique `computedrole` string as the return value for interoperability testing purposes, such as `&lt;video&gt; -&gt; "html-video"` in [[HTML-AAM]]. However, authors MUST NOT use any non-concrete `computedrole` string in the `role` attribute, and User Agents MUST ignore any non-concrete role token.</p>
```

Reply:
Or
```suggestion
      <p>When host language elements do not have an exact or equivalent mapping to a concrete role (valid and non-abstract), the related Accessibilty API Mapping extension specification MAY specify a unique `computedrole` string as the return value for interoperability testing purposes, such as `&lt;video&gt; -&gt; "html-video"` in [[HTML-AAM]]. However, authors MUST NOT use any non-concrete `computedrole` string in the `role` attribute, and User Agents MUST ignore any non-concrete role token.</p>
```

Reply:
yes, you're right, gotta take into account the non-ARIA ones.  

I prefer "mapping to a valid, non-abstract role" then.

Reply:
if we removed the usage of concrete role shouldn't we also remove the 2 usages of non-concrete roles?

Reply:
I should've just defined "concrete."

Reply:
https://github.com/w3c/core-aam/pull/176#discussion_r1206113499